modules/post/multi/manage/upload_exec.rb

```diff
@@ -0,0 +1,52 @@
+##
+# This module requires Metasploit: http//metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+
+  def initialize(info={})
+    super( update_info( info,
+      'Name'          => 'Upload and Execute',
+      'Description'   => %q{ Push a file and execute it },
+      'License'       => MSF_LICENSE,
+      'Author'        => [ 'egypt'],
+      'Platform'      => [ 'win','linux','osx' ],
```

Comment:
Hmm. I tested with a python stager from msfvenom and it worked for me (after adding a shebang line). I'll try with an elf